iDiff/differs/differs.go

```diff
@@ -28,30 +30,28 @@ func Diff(arg1, arg2, differ string, json bool) (string, error) {
 func specificDiffer(f func(string, string, bool) (string, error), img1, img2 string, json bool) (string, error) {
 	var buffer bytes.Buffer
 	validDiff := true
-	jsonPath1, dirPath1, err := utils.ImageToDir(img1)
+	imgPath1, err := utils.ImageToFS(img1)
 	if err != nil {
 		buffer.WriteString(err.Error())
 		validDiff = false
 	}
-	jsonPath2, dirPath2, err := utils.ImageToDir(img2)
+	imgPath2, err := utils.ImageToFS(img2)
 	if err != nil {
 		buffer.WriteString(err.Error())
 		validDiff = false
 	}
 
 	var diff string
 	if validDiff {
-		output, err := f(jsonPath1, jsonPath2, json)
+		output, err := f(imgPath1, imgPath2, json)
 		if err != nil {
 			buffer.WriteString(err.Error())
 		}
 		diff = output
 	}
 
-	errStr := remove(dirPath1, true, "")
-	errStr = remove(dirPath2, true, errStr)
-	errStr = remove(jsonPath1, false, errStr)
-	errStr = remove(jsonPath2, false, errStr)
+	errStr := remove(imgPath1, true, "")
+	errStr = remove(imgPath2, true, errStr)
```

Comment:
It seems a little strange to me to pass in the existing `errStr` in order to do the concatenation inside this helper method. Can you remove that param to `remove()` and then combine the error strings in this method?